src/Types/Harness/Repository/Repository.php

```diff
@@ -75,7 +75,13 @@ private function importPackagesFromSources()
                 continue;
             }
 
-            $this->packages = array_merge($this->packages, json_decode(file_get_contents($source['url']), true));
+            $packages = json_decode(file_get_contents($source['url']), true, JSON_THROW_ON_ERROR);
+
+            if ($packages === null) {
+                throw new Exception(sprintf('Response of %s can\'t be decoded as json.', $source['url']));
+            }
```

Comment:
@a-ast i don't think we need the NULL check AND the flag?

Reply:
but in anycase, I since added this sevice:

https://github.com/my127/workspace/blob/feature/multiple-harness-layers/src/File/JsonLoader.php

which handles loading JSON in a way that can be better unit tested, so might make sense to refactor this class to use that....